dask_sql/physical/rel/custom/ml_classes.py

```diff
@@ -0,0 +1,381 @@
+        "Perceptron": "sklearn.linear_model.Perceptron",
+        "RidgeClassifier": "sklearn.linear_model.RidgeClassifier",
+        "RidgeClassifierCV": "sklearn.linear_model.RidgeClassifierCV",
+        "SGDClassifier": "sklearn.linear_model.SGDClassifier",
+        "SGDOneClassSVM": "sklearn.linear_model.SGDOneClassSVM",
+        "LinearRegression": "sklearn.linear_model.LinearRegression",
+        "Ridge": "sklearn.linear_model.Ridge",
+        "RidgeCV": "sklearn.linear_model.RidgeCV",
+        "SGDRegressor": "sklearn.linear_model.SGDRegressor",
+        "ElasticNet": "sklearn.linear_model.ElasticNet",
+        "ElasticNetCV": "sklearn.linear_model.ElasticNetCV",
+        "Lars": "sklearn.linear_model.Lars",
+        "LarsCV": "sklearn.linear_model.LarsCV",
+        "Lasso": "sklearn.linear_model.Lasso",
+        "LassoCV": "sklearn.linear_model.LassoCV",
+        "LassoLars": "sklearn.linear_model.LassoLars",
+        "LassoLarsCV": "sklearn.linear_model.LassoLarsCV",
+        "LassoLarsIC": "sklearn.linear_model.LassoLarsIC",
+        "OrthogonalMatchingPursuit": "sklearn.linear_model.OrthogonalMatchingPursuit",
+        "OrthogonalMatchingPursuitCV": "sklearn.linear_model.OrthogonalMatchingPursuitCV",
+        "ARDRegression": "sklearn.linear_model.ARDRegression",
+        "BayesianRidge": "sklearn.linear_model.BayesianRidge",
+        "MultiTaskElasticNet": "sklearn.linear_model.MultiTaskElasticNet",
+        "MultiTaskElasticNetCV": "sklearn.linear_model.MultiTaskElasticNetCV",
+        "MultiTaskLasso": "sklearn.linear_model.MultiTaskLasso",
+        "MultiTaskLassoCV": "sklearn.linear_model.MultiTaskLassoCV",
+        "HuberRegressor": "sklearn.linear_model.HuberRegressor",
+        "QuantileRegressor": "sklearn.linear_model.QuantileRegressor",
+        "RANSACRegressor": "sklearn.linear_model.RANSACRegressor",
+        "TheilSenRegressor": "sklearn.linear_model.TheilSenRegressor",
+        "PoissonRegressor": "sklearn.linear_model.PoissonRegressor",
+        "TweedieRegressor": "sklearn.linear_model.TweedieRegressor",
+        "GammaRegressor": "sklearn.linear_model.GammaRegressor",
+        "PassiveAggressiveRegressor": "sklearn.linear_model.PassiveAggressiveRegressor",
+        # sklearn.manifold: Manifold Learning
+        "Isomap": "sklearn.manifold.Isomap",
+        "LocallyLinearEmbedding": "sklearn.manifold.LocallyLinearEmbedding",
+        "MDS": "sklearn.manifold.MDS",
+        "SpectralEmbedding": "sklearn.manifold.SpectralEmbedding",
+        "TSNE": "sklearn.manifold.TSNE",
+        # sklearn.mixture: Gaussian Mixture Models
+        "BayesianGaussianMixture": "sklearn.mixture.BayesianGaussianMixture",
+        "GaussianMixture": "sklearn.mixture.GaussianMixture",
+        # sklearn.model_selection: Model Selection
+        "GroupKFold": "sklearn.model_selection.GroupKFold",
+        "GroupShuffleSplit": "sklearn.model_selection.GroupShuffleSplit",
+        "KFold": "sklearn.model_selection.KFold",
```

Comment:
There's probably a lot of mappings here that aren't even compatible with `model_class` or `experiment_class`.

I pretty much just listed every class available with sklearn and cuml and allow it to be mapped no matter what, since these are just string mappings and it has to error eventually anyways if it's not compatible with `CREATE MODEL`/`CREATE EXPERIMENT`. But let me know if there are chunks of mappings that I should just remove.

Reply:
My preference would be to generate this mapping dynamically for a few reasons:

- as scikit-learn updates, it's very likely that many of these classes will get shuffled around, which will end up breaking CI until a PR is made to manually update things
- though this mapping currently works on the scikit-learn version getting pulled in our python>=3.9 testing, we don't actually know how many versions this mapping holds up for between 1.0.0 and 1.2.1

It looks like for scikit-learn, we should be able to do something like

```python
from sklearn.utils import all_estimators

estimators = {k: v.__module__ + "." + v.__qualname__ for k,v in all_estimators()}
```

To grab its estimators dynamically; I'm not sure if the other packages have something similar to this, but would be interested in if it's possible to generalize the code used for [`all_estimators`](https://github.com/scikit-learn/scikit-learn/blob/e2e705021eb6c9f23f0972f119b56e37cd7567ef/sklearn/utils/discovery.py#L17-L127).

Reply:
I didn't know about this, cool! LightGBM and XGBoost don't seem to have a similar function, but they have a much smaller number of estimators anyway where I wouldn't expect them to be prone to changing.

I couldn't find the function in cuML, but I opened https://github.com/rapidsai/cuml/issues/5162 as a possibility for the future. In the meantime, I kept the `gpu_classes` dict as is.